### PR TITLE
Move serialization from remixer_storage to remixer_core

### PR DESCRIPTION
**Pull request by miguelandres**
This allows for cleaner integration with Core, and having some sensible defaults that don't need to be explicitly set.

This is the last PR before #50 is refactored to the comments on it.

### CODE REVIEW

File: remixer_core/build.gradle [hunk 1]
```diff
@@ -28,6 +28,8 @@ jacocoTestReport {
 }
 
 dependencies {
+  compile 'com.google.code.gson:gson:2.7'
   testCompile 'junit:junit:4.12'
   testCompile 'org.mockito:mockito-core:1.9.5'
+
```
Comment (pingpongboss) [inline remixer_core/build.gradle]: remove?

File: remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java [hunk 2]
```diff
@@ -20,6 +26,10 @@
   private final Class valueClass;
```
Comment (pingpongboss) [inline remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java]: should this be `Class<T>`?

File: remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java [hunk 3]
```diff
@@ -71,15 +89,25 @@ public String getName() {
   public Class getValueClass() {
```
Comment (pingpongboss) [inline remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java]: and this?

File: remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java [hunk 4]
```diff
@@ -28,9 +38,17 @@
   private final Map<Class<? extends RemixerItem>, Integer> layoutIdForRemixerItemType =
       new HashMap<>();
 
-  public DataType(String name, Class valueClass) {
+  /**
+   * Constructs a datatype with the given {@code name}, that takes values of type {@code valueClass}
+   * and uses {@code converter} to serialize.
+   *
+   * <p>Note {@code converter} has a {@link ValueConverter#dataType} field that must be initialized
+   * to the same as {@code name}.
```
Comment (pingpongboss) [inline remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java:47]: You should check this precondition if it's important. But why?

File: remixer_core/src/main/java/com/google/android/libraries/remixer/serialization/StoredVariable.java [hunk 5]
```diff
@@ -122,12 +201,13 @@ public int hashCode() {
...
     StoredVariable storedVariable = null;
-    for (SupportedDataType type : SupportedDataType.values()) {
+    for (DataType type : Remixer.getInstance().getRegisteredDataType()) {
       try {
-        storedVariable = type.getValueConverter().fromRemixerItem(item);
+        storedVariable = type.getConverter().fromRemixerItem(item);
         break;
       } catch (IllegalArgumentException ex) {
         // Don't do anything, this just wasn't the right data type.
+        storedVariable = null;
```
Comment (pingpongboss) [inline remixer_core/src/main/java/com/google/android/libraries/remixer/serialization/StoredVariable.java:210]: Did you add this for a reason? `storedVariable` will never be non-null here, right?

**Review by pingpongboss — APPROVED** [review]

File: remixer_core/build.gradle (hunk 1 above)
Comment (miguelandres) [reply remixer_core/build.gradle]: Done

File: remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java (hunk 2 above)
Comment (miguelandres) [reply remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java]: Right!

File: remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java (hunk 4 above)
Comment (miguelandres) [reply remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java:47]: Done :). It is important for parsing/exporting.

File: remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java (hunk 3 above)
Comment (miguelandres) [reply remixer_core/src/main/java/com/google/android/libraries/remixer/DataType.java]: Done.

File: remixer_core/src/main/java/com/google/android/libraries/remixer/serialization/StoredVariable.java (hunk 5 above)
Comment (miguelandres) [reply remixer_core/src/main/java/com/google/android/libraries/remixer/serialization/StoredVariable.java:210]: pmd complains for not having anything in a catch, not doing anything is the right thing in this case though.

**Pull request merged**